src/char/char_clif.c

```diff
@@ -523,6 +527,36 @@ int chclif_parse_char_delete2_req(int fd, struct char_session_data* sd) {
 	return 1;
 }
 
+/**
+ * Check char deletion code
+ * @param sd
+ * @param delcode E-mail or birthdate
+ * @param flag Delete flag
+ * @return 1:Success, 0:Failure
+ **/
+static int chclif_delchar_check(struct char_session_data *sd, char *delcode, uint8 flag) {
+	// E-Mail check
+	if (flag&1) {
+		if (strcmp(delcode, sd->email) && //email does not matches and
+			(strcmp("a@a.com", sd->email) || //it is not default email, or
+			(strcmp("a@a.com", delcode) && strcmp("", delcode)) //email sent does not matches default
+			))
```

Comment:
yea, bit confused with that code. I wanted just check
 * 1. Is it match
 * 2. Not default email
for empty email/birthdate, must always be checked first, tbh.